api/runner/runner.go

```diff
@@ -29,6 +29,7 @@ type Runner struct {
 	flog         FuncLogger
 	availableMem int64
 	usedMem      int64
+	Tasks        chan task.Request
```

Comment:
I know I am being a sort of blocker. In the end, I will support you on whatever you decide. I am just trying to spark reasoning about changes.

> Same for other comments that would decentralize things and keep it more simple.

In my mind, we would benefit a lot of having smaller components that are easier to reason about, than one big component with lots of not-so-obvious internal relationships.

The downside, is that you have more dots to connect. (See comment below about that).